Extensions/ISearchContextExtensions.cs

```diff
@@ -0,0 +1,75 @@
+﻿using OpenQA.Selenium;
+using OpenQA.Selenium.Support.UI;
+using System;
+using OpenQA.Selenium.Interactions;
+
+
+namespace Extensions
+{
+    public static class ISearchContextExtensions
+    {
+        public static IWebElement WaitAndFindElement(this ISearchContext iSearchContext, By by)
+        {
+            try
+            {
+                var wait = new DefaultWait<ISearchContext>(iSearchContext);
+                wait.Timeout = TimeSpan.FromSeconds(15);
+                return wait.Until(ctx =>
+                {
+                    var elem = ctx.FindElement(by);
+                    wait.IgnoreExceptionTypes(typeof(NoSuchElementException), typeof(StaleElementReferenceException));
+                    if (!elem.Enabled || !elem.Displayed)
+                    {
+                        return null;
+                    }
+
+                    return elem;
```

Comment:
instead, you can write 
```suggestion
                    if (elem.Enabled && elem.Displayed)
                    {
                        return elem;
                    }

                    return null;
```

which makes more sense because you're waiting for the element to be shown, and not for the element to not be showing